crates/topos/src/components/node/services/process.rs

```diff
@@ -0,0 +1,137 @@
+use crate::config::sequencer::SequencerConfig;
+use crate::config::tce::TceConfig;
+use crate::edge::CommandConfig;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::time::Duration;
+use thiserror::Error;
+use tokio::{spawn, sync::mpsc, task::JoinHandle};
+use tokio_util::sync::CancellationToken;
+use topos_p2p::config::NetworkConfig;
+use topos_sequencer::SequencerConfiguration;
+use topos_tce::config::{AuthKey, StorageConfiguration, TceConfiguration};
+use topos_tce_transport::ReliableBroadcastParams;
+use topos_wallet::SecretManager;
+use tracing::{debug, error, info};
+
+use crate::config::genesis::Genesis;
+
+#[derive(Error, Debug)]
+pub enum Errors {
+    #[error("Failure on the TCE")]
+    TceFailure,
+    #[error("Failure on the Sequencer")]
+    SequencerFailure,
+    #[error("Failure on the Edge")]
```

Comment:
Maybe "THING failure" is easier to read than "Failure on THING"?

Reply:
Maybe we should use something like: `Edge error: {0}`